staging/src/k8s.io/kubectl/pkg/drain/drain.go

```diff
@@ -277,6 +277,9 @@ func (d *Helper) evictPods(pods []corev1.Pod, policyGroupVersion string, getPodF
 				} else if apierrors.IsTooManyRequests(err) {
 					fmt.Fprintf(d.ErrOut, "error when evicting pod %q (will retry after 5s): %v\n", pod.Name, err)
 					time.Sleep(5 * time.Second)
+				} else if apierrors.IsForbidden(err) {
```

Comment:
We need to check for the special forbidden cause (HasStatusCause namespace terminating) and then we need to time.Sleep(5 * time.Second) and refresh the pod (basically we are waiting for the pod to be marked deleting).  You'd have to get @michaelgugino to suggest what the best way to reflect that is.

To rephrase:

1. If an eviction attempt fails because the namespace is being deleted
2. start waiting for the pod to be marked for deletion (deletionTimestamp)
3. when it marked for deletion, waitForDelete

We have to wait until the pod is marked for deletion or we could exit drain before the pod has a chance to shut down gracefully.

Reply:
If we break here... then on line 304 there is a waitForDelete, which updates the pod.

Reply:
That middle step on waiting for a pod to be marked for deletion might not be needed.

Reply:
I'm worried that Forbidden is too broad, I think we should identify the cause as @smarterclayton is suggesting, or just ignore the reason and check that the pod has a deletion timestamp (I think the latter is simpler and more generalized if something like this crops up in the future).

While the patch as implemented would indeed waitForDelete, I think we're breaking the intent of the code somewhat and we'll have a surprise for future updates later.  Right now, the implied intent of this function is that pod in question has been evicted successfully (eg, marked for deletion), is missing (already deleted) or error.  Adding a new nebulous state of "Probably will get deleted by someone else so we can ignore it" is less than ideal as it might break assumptions later in the code (albeit not today, but potentially in the future).